app/apollo/resolvers/group.js

```diff
@@ -485,18 +529,14 @@ const groupResolvers = {
       const user = whoIs(me);
 
       try {
-        logger.info({ req_id, user, org_id, groupUuids, clusterId }, `${queryName} validating`);
-
-        await validAuth(me, org_id, ACTIONS.MANAGE, TYPES.GROUP, queryName, context);
-
         validateString( 'org_id', org_id );
         validateString( 'clusterId', clusterId );
         groupUuids.forEach( value => validateString( 'groupUuids', value ) );
 
-        const groups = await models.Group.find({ org_id, uuid: { $in: groupUuids } });
-        if (groups.length != groupUuids.length) {
-          throw new NotFoundError(context.req.t('One or more of the passed group uuids were not found'));
-        }
+        let groups = await getAllowedResources(me, org_id, ACTIONS.MANAGE, TYPES.GROUP, queryName, context, null, groupUuids);
```

Comment:
Does this code need to verify that the user has at least READ permission on the _cluster_ as well?
To do so, it will need to query the cluster from the DB first, so it can get the name for auth check.  This will combo well with later comment that it needs to retrieve the cluster for passing to the pluginContext, though keep in mind that the cluster is not required to exist for this function to succeed (a somewhat questionable design decision, but we should not change the behavior with this PR beyond enforcing authorization better.

Reply:
added cluster db retrieval and validAuth() call here in editClusterGroups

Reply:
I left out the NotFoundError/validation error that is normall in these calls since we want it to continue either way

Reply:
https://cloud.ibm.com/docs/satellite?topic=satellite-iam-service-access#service-access-cluster-grp I believe we need to keep using the MANAGE permission for editClusterGroups, take a look at the table of the cluster group roles here.

Reply:
Resolved